From 44ad7f9a945d325677066fbf0d984df374587170 Mon Sep 17 00:00:00 2001
From: Michal-Szczepaniak <michal_sz13@o2.pl>
Date: Fri, 9 Mar 2018 22:12:38 +0100
Subject: [PATCH 1/2] Fix splitting if there is more than one separator

---
 hybris/common/n/linker.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hybris/common/n/linker.cpp b/hybris/common/n/linker.cpp
index 213b6c36..d0c560a8 100644
--- a/hybris/common/n/linker.cpp
+++ b/hybris/common/n/linker.cpp
@@ -210,7 +210,7 @@ static const char* const kAsanDefaultLdPaths[] = {
 std::vector<std::string> split(const std::string &text, const std::string &sep) {
   std::vector<std::string> tokens;
   std::size_t start = 0, end = 0;
-  while ((end = text.find(sep, start)) != std::string::npos) {
+  while ((end = text.find_first_of(sep, start)) != std::string::npos) {
     tokens.push_back(text.substr(start, end - start));
     start = end + 1;
   }

From c214b8f59c57d147eda23082928122c8d7cfc2b9 Mon Sep 17 00:00:00 2001
From: Michal-Szczepaniak <michal_sz13@o2.pl>
Date: Fri, 9 Mar 2018 23:53:46 +0100
Subject: [PATCH 2/2] Add HYBRIS_LD_SHIM_LIBS support

---
 hybris/common/n/linker.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/hybris/common/n/linker.cpp b/hybris/common/n/linker.cpp
index d0c560a8..5f775128 100644
--- a/hybris/common/n/linker.cpp
+++ b/hybris/common/n/linker.cpp
@@ -4633,9 +4633,11 @@ extern "C" void android_linker_init(int sdk_version, void* (*get_hooked_symbol)(
 
   const char* ldpath_env = nullptr;
   const char* ldpreload_env = nullptr;
+  const char* ldshim_libs_env = nullptr;
   if (!getauxval(AT_SECURE)) {
     ldpath_env = getenv("HYBRIS_LD_LIBRARY_PATH");
     ldpreload_env = getenv("HYBRIS_LD_PRELOAD");
+    ldshim_libs_env = getenv("HYBRIS_LD_SHIM_LIBS");
   }
 
   if (DEFAULT_HYBRIS_LD_LIBRARY_PATH)
@@ -4643,6 +4645,7 @@ extern "C" void android_linker_init(int sdk_version, void* (*get_hooked_symbol)(
   else
     parse_LD_LIBRARY_PATH(ldpath_env);
   parse_LD_PRELOAD(ldpreload_env);
+  parse_LD_SHIM_LIBS(ldshim_libs_env);
 
   if (sdk_version > 0)
     set_application_target_sdk_version(sdk_version);
